docs/odcs.rst

```diff
@@ -24,84 +24,188 @@ idea is to not require the user to provide a yum repo url when building
+``beta``, and ``unsigned``. Each one of those intents is then mapped to a list
+of signing keys. These signing keys are then used during compose creation. The
+packages to be included must have been signed by either one of the keys listed.
+In the example above, the intents could be mapped to the following keys::
+
+    # Only include packages that have been signed by "my-release-key"
+    release -> my-release-key
+    # Include packages that have been signed by either "my-beta-key" or
+    # "my-release-key"
+    beta -> my-beta-key, my-release-key
+    # Do not check signature of packages - may include unsigned packages
+    unsigned -> <empty>
+
+*This mapping will be defined in reactor-config-secret which will allow updates
+to be reflected instantly for automatically triggered builds from an existing
+BuildConfig.*
+
+The definition of intents also describes the restrictive order, which will be
+enforced when building layered images. For instance, consider the case of two
+images, X and Y. Y uses X as its parent image (FROM X). If image X was built
+with "beta" intent, image Y's intent can only be "beta" or "unsigned". If the
+repo for image Y has it configured to use "release" intent, this value will be
+downgraded to "beta" at build time.
+
+In the case where multiple composes are used, the least restrictive intent is
+used. Continuing with our previous signing intent example, let's say a container
+image build request uses two composes. Compose 1 was generated with no signing
+keys provided, and compose 2 was generated with "my-release-key". In this case,
+the intent is "unsigned".
+
+Because compose IDs can be passed in to OSBS, a mechanism will be in place to
+classify the intent of an existing compose. This will be done by inspecting the
+signing keys used for generating the compose and performing a reverse mapping to
+determine the intent. If a match cannot be determined, intent will be set to
+``None``, which is less restrictive than any defined signing intents.
+
+The ``signing_intent`` specified in ``container.yaml`` can be overridden with
+the build parameter of same name. This particular parameter will be ignored for
+autorebuilds. The value in ``container.yaml`` should always be used in that
+case. Note that the signing intent used by the compose of parent image is still
+enforced.
+
+Enabling ODCS Integration
+"""""""""""""""""""""""""
+
+The first step in enabling ODCS integration is by configuring the OSBS
+environment to use a specific ODCS server via the ``odcs_*`` configuration
+parameters.
+
+Next, ``container.yaml`` must be created and it must include the *compose*
+section. See example further down. This file must then be committed and pushed
+to git repository. Simply building the container image as usual will trigger
+ODCS interaction.
+
+
+ODCS Compose Checks
+"""""""""""""""""""
+
+When an ODCS compose is passed in as an argument, OSBS will perform some checks
```

Comment:
Would it be better to say 'ODCS compose ID'?

Reply:
Yes. Fixed.